File: app/src/main/java/com/kieran/winnipegbusbackend/Stop.java
package com.kieran.winnipegbusbackend;

import java.io.Serializable;

public class Stop implements Serializable {
    public final static String STOP_NAME_TAG = "name";
    public final static String STOP_NUMBER_TAG = "number";
    public final static String STOP_TAG = "stop";
    protected String stopName;
    protected int stopNumber;

    public Stop() {
        stopName = "";
        stopNumber = 0;
    }

    public Stop(String stopName, int stopNumber) {
        this.stopName = stopName;
        this.stopNumber = stopNumber;
    }

    public Stop(int stopNumber) {
        this.stopNumber = stopNumber;
        stopName = "";
    }

    public Stop(Stop stop) {
        stopName = stop.getName();
        stopNumber = stop.getNumber();
    }

    public String getName() {
        return stopName;
    }

    public int getNumber() {
        return stopNumber;
    }

    @Override
    public String toString() {
        return Integer.toString(stopNumber) + " " + stopName;
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/Stop.kt
package com.kieran.winnipegbusbackend

import java.io.Serializable

open class Stop : Serializable {
    var name: String
        protected set
    var number: Int = 0
        protected set

    constructor() {
        name = ""
        number = 0
    }

    constructor(stopName: String, stopNumber: Int) {
        this.name = stopName
        this.number = stopNumber
    }

    constructor(stopNumber: Int) {
        this.number = stopNumber
        name = ""
    }

    constructor(stop: Stop) {
        name = stop.name
        number = stop.number
    }

    override fun toString(): String {
        return Integer.toString(number) + " " + name
    }
}
